### PR TITLE
Change Model to have new classes: Event and Task as subclasses of Activity

### PR DESCRIPTION
**Pull request by YuanQLLer**
Implemented the event and task class.
I will change the ui and storage later to make it work.

I will fix the format of test part latter.

### CODE REVIEW

**Comment by karenfrilya97** [comment]
#34

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -60,47 +60,47 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 
 Format: `help`
 
-=== Adding a person: `add`
```
Comment (Kyomian) [inline docs/UserGuide.adoc]: The User Guide must be changed completely. I will change it later, so this UG code is ok.

File: docs/team/johndoe.adoc [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 == Overview
 
-AddressBook - Level 4 is a desktop address book application used for teaching Software Engineering principles. The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java, and has about 10 kLoC.
+AddressBook - Level 4 is a desktop remark book application used for teaching Software Engineering principles. The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java, and has about 10 kLoC.
```
Comment (Kyomian) [inline docs/team/johndoe.adoc]: @YuanQLLer Change this to:
CLIndar is a desktop application that helps university computing students to manage their tasks and events. The user interacts with it using a CLI. It has a GUI created with JavaFX, and is written in Java.

File: src/main/java/seedu/address/commons/events/model/DeskBoardChangedEvent.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package seedu.address.commons.events.model;
+
+import seedu.address.commons.events.BaseEvent;
+import seedu.address.model.ReadOnlyDeskBoard;
+
+/** Indicates the DeskBoard in the model has changed*/
```
Comment (Kyomian) [inline src/main/java/seedu/address/commons/events/model/DeskBoardChangedEvent.java]: Please add:
//@@author YuanQQLer 
above /** Indicates the DeskBoard in the model has changed*/

File: src/main/java/seedu/address/commons/events/ui/ActivityPanelSelectionChangedEvent.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package seedu.address.commons.events.ui;
+
+import seedu.address.commons.events.BaseEvent;
+import seedu.address.ui.ActivityCard;
+
+/**
```
Comment (Kyomian) [inline src/main/java/seedu/address/commons/events/ui/ActivityPanelSelectionChangedEvent.java]: Add
//@@author YuanQQLer

File: src/main/java/seedu/address/model/DeskBoard.java [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+import javafx.collections.ObservableList;
+import seedu.address.model.activity.Activity;
+import seedu.address.model.activity.UniqueActivityList;
+import seedu.address.model.activity.exceptions.ActivityNotFoundException;
+import seedu.address.model.activity.exceptions.DuplicateActivityException;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.tag.UniqueTagList;
+
+/**
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/DeskBoard.java]: Add
//@@author YuanQQLer

File: src/main/java/seedu/address/model/Model.java [hunk 6]
```diff
@@ -3,46 +3,47 @@
...
 import javafx.collections.ObservableList;
-import seedu.address.model.person.Person;
-import seedu.address.model.person.exceptions.DuplicatePersonException;
-import seedu.address.model.person.exceptions.PersonNotFoundException;
+import seedu.address.model.activity.Activity;
+import seedu.address.model.activity.exceptions.ActivityNotFoundException;
+import seedu.address.model.activity.exceptions.DuplicateActivityException;
+
 
 /**
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/Model.java]: Add
//@@author YuanQQLer

File: src/main/java/seedu/address/model/ModelManager.java [hunk 7]
```diff
@@ -11,10 +11,10 @@
...
 import seedu.address.commons.core.ComponentManager;
 import seedu.address.commons.core.LogsCenter;
-import seedu.address.commons.events.model.AddressBookChangedEvent;
-import seedu.address.model.person.Person;
-import seedu.address.model.person.exceptions.DuplicatePersonException;
-import seedu.address.model.person.exceptions.PersonNotFoundException;
+import seedu.address.commons.events.model.DeskBoardChangedEvent;
+import seedu.address.model.activity.Activity;
+import seedu.address.model.activity.exceptions.ActivityNotFoundException;
+import seedu.address.model.activity.exceptions.DuplicateActivityException;
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/ModelManager.java]: Add
//@@author YuanQQLer

File: src/main/java/seedu/address/model/ReadOnlyDeskBoard.java [hunk 8]
```diff
@@ -1,19 +1,19 @@
 package seedu.address.model;
 
 import javafx.collections.ObservableList;
-import seedu.address.model.person.Person;
+import seedu.address.model.activity.Activity;
 import seedu.address.model.tag.Tag;
 
 /**
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/ReadOnlyDeskBoard.java]: Add
//@@author YuanQQLer

File: src/main/java/seedu/address/model/activity/Activity.java [hunk 9]
```diff
@@ -10,27 +10,27 @@
 import seedu.address.model.tag.UniqueTagList;
 
 /**
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/activity/Activity.java]: Add
//@@author YuanQQLer

File: src/main/java/seedu/address/model/activity/DateTime.java [hunk 10]
```diff
@@ -0,0 +1,82 @@
+package seedu.address.model.activity;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/activity/DateTime.java]: Add
//@@author YuanQQLer

File: src/main/java/seedu/address/model/activity/Event.java [hunk 11]
```diff
@@ -0,0 +1,120 @@
+package seedu.address.model.activity;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+
+
+// @@YuanQLLer
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/activity/Event.java]: Wrong Format
//@@author YuanQQLer

File: src/main/java/seedu/address/model/activity/Location.java [hunk 12]
```diff
@@ -0,0 +1,57 @@
+package seedu.address.model.activity;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/activity/Location.java]: Add
//@@author YuanQQLer

File: src/main/java/seedu/address/model/activity/Remark.java [hunk 13]
```diff
@@ -0,0 +1,54 @@
+package seedu.address.model.activity;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/activity/Remark.java]: Add 
//@@author YuanQQLer

File: src/main/java/seedu/address/model/activity/Task.java [hunk 14]
```diff
@@ -0,0 +1,95 @@
+package seedu.address.model.activity;
+
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+
+// @@YuanQLLer
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/activity/Task.java]: Wrong Format
//@@author YuanQQLer

File: src/main/java/seedu/address/model/activity/UniqueActivityList.java [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.model.activity.exceptions.ActivityNotFoundException;
+import seedu.address.model.activity.exceptions.DuplicateActivityException;
+
+
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/activity/UniqueActivityList.java]: Add
//@@author YuanQQLer

File: src/main/java/seedu/address/model/activity/exceptions/ActivityNotFoundException.java [hunk 16]
```diff
@@ -0,0 +1,6 @@
+package seedu.address.model.activity.exceptions;
+
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/activity/exceptions/ActivityNotFoundException.java]: Add
//@@author YuanQQLer

File: src/main/java/seedu/address/model/activity/exceptions/DuplicateActivityException.java [hunk 17]
```diff
@@ -0,0 +1,12 @@
+package seedu.address.model.activity.exceptions;
+
+import seedu.address.commons.exceptions.DuplicateDataException;
+
```
Comment (Kyomian) [inline src/main/java/seedu/address/model/activity/exceptions/DuplicateActivityException.java]: Add
//@@author YuanQQLer

**Pull request merged**